### ISSUE TITLE
Reproduction in Vanilla setup showing poor performances.

### ISSUE THREAD

**Issue by GurvanR** [pasquini-dario/LLMmap#2]
Hello, 
I'm a PhD student looking for comparing your fingerprinting method with others.
Thank you very much for updating your repo recently, it's very helpful for our evaluating purpose. However, we're having an issue. We want to use LLMmap in a specific setup that is quite vanilla: T=1.0, freq_pen =0.0 no system prompts etc. and with 22 of your LLMs. We use vllm to make the inferences with
```
'dtype: "bfloat16",
'top_k': 50,
'max_tokens': 500,
'temperature':1.0
```

We plotted the global accuracy as a function of the nb_of_queries, using the default_queries set.
The performances we got were terrible compared to what you have in your paper (see figure provided).
Would you have any idea of why this happened? The code is also attached.

Also, why the classifier model provided don't take less than 8 queries? By reading the paper, I thought the model $f$ could take different number of traces while trained on a fixed one. Then to simulate it, we "padded" the rest of required queries with empty ones. 

Second surprise, the performances were better with 7 queries than 8.

Thank you very much for your attention and your help. 
Gurvan

<img width="1200" height="800" alt="Image" src="https://github.com/user-attachments/assets/514642c5-a8a2-4db5-ab79-f3c8c7b998c3" />

[LLMmap_Reproduction_issue.zip](https://github.com/user-attachments/files/21852717/LLMmap_Reproduction_issue.zip)

**Comment by pasquini-dario** [comment]
Hello. Yep. That is strange.

Thanks for the code. Let me look into it.

>  Also, why the classifier model provided don't take less than 8 queries? By reading the paper, I thought the model  f could take different number of traces while trained on a fixed one. Then to simulate it, we "padded" the rest of required queries with empty ones.

Yes, that feature should not be working on the model trained with the last ./train script.  In the original, training inputs for the model were sampled randomly rather than providing always the 8 traces (as happens now).

If possible, could you re-run by setting top_k to the default value and see what happens?

**Comment by pasquini-dario** [comment]
A quick observation: Giving a look to traces.csv, it seems that some models produce broken outputs (e.g., line 11). Might it be that no chat template (e.g., https://huggingface.co/learn/llm-course/en/chapter11/2) is used to query those models?  Another possible issue is that the sampling hyperparameters are too extreme for those specific models (I remember we struggled with that too, and we had to reduce the range for frequency_penalty for some LLMs to avoid breaking them). Let me know.

**Comment by GurvanR** [comment]
First of all, thank you very much for getting back to us so quickly.

Ok, so the model provided currently only supports the 8 (default) queries and no less.

The top_k value is considered default in https://huggingface.co/docs/transformers/main_classes/text_generation, you mean rerun without setting it to 50 so that maybe the value will differ from an LLM to another?

Regarding the broken outputs, yes, that is also our main suspect. We will need to look into this part further. Did you use chat templates? we thought this was not necessary for a single input/output interaction. To which sampling parameters are you refering to? We try to be as "standard" as possible, by following default values of this https://huggingface.co/docs/transformers/main_classes/text_generation.

And did you use vllm or standard transformers for doing inferences?

**Comment by pasquini-dario** [comment]
> Regarding the broken outputs, yes, that is also our main suspect. We will need to look into this part further. Did you use chat templates?

Yes, usually, to work properly, instruction-tuned LLMs need to receive instructions following the template/roles format they have been trained to use; otherwise, they might just produce random junk. As far as I know, using a chat template is the correct way (and what happens under the hood when evoking transformers.pipeline).

Give a look to the make_prompt methods in LLMma.llm for details.

> And did you use vllm or standard transformers for doing inferences?
standard transformers.

let me know if that solves.

**Comment by GurvanR** [comment]
Yes, sure, thank you for raising this crucial point! 
We'll let you know.

**Comment by GurvanR** [comment]
We have been able to set the chat templates on 17 LLMs and got this result. So only the 8 queries accuracy can reflects fairly the classifier $f$ as it is not intended to take less than the 8 default queries. The performances are much much better than without chat templates, though they remain lower than in the paper..

<img width="1200" height="800" alt="Image" src="https://github.com/user-attachments/assets/f0a01c36-b8c7-441f-a795-4086549367b3" />

[LLMmap_Reproduction_issue_wCT.zip](https://github.com/user-attachments/files/21969725/LLMmap_Reproduction_issue_wCT.zip)

**Comment by pasquini-dario** [comment]
Umm. Let me see.

**Comment by pasquini-dario** [comment]
You used the same code as in the previous zip file, right?

**Comment by GurvanR** [comment]
> You used the same code as in the previous zip file, right?

yes that's right, the traces have been updated, they come from generation with chat template.

**Comment by pasquini-dario** [comment]
OK. Give me some time to look into it in detail. If it is not there, could you share the code you used to generate the traces?

**Comment by GurvanR** [comment]
No I haven't given it yet, here it is, note that the parameters used are the one described in my first message. 
Here is how it works: my main function initializes a vLLM object from the vLLM_Classes.py file and then proceeds to generation with .infer() method. I don't give you my main as it contains a lot of things that I don't think useful for what we're trying to understand.
It is a bit messy sorry for that I hope it'll be understandable enough.

Thank you again for your investigation!!

[Inference_code.zip](https://github.com/user-attachments/files/21988262/Inference_code.zip)

**Comment by pasquini-dario** [comment]
@GurvanR  , sorry for the delay; working on the issue now.  Could you share the file containing the queries run to generate the traces (not sure I have it)? I mean the file you use to fill up the data frame here:

```raw_prompts = [str(prompt) for prompt in self.dataframe['Question']]``` 

in General_LLM_Class.py.

I just realized I am using a set of queries slightly different from the one reported in the paper for this model (see confs/queries/default.json), and I want to make sure there is no mismatch before going deeper into the code.

**Comment by GurvanR** [comment]
Hello Dario, no problem, I used the default queries of your repo: confs/queries/default.json.
